plugin/server/datastore-sqlite/sqlite.go

```diff
@@ -847,29 +877,46 @@ func (ds *sqlitePlugin) sortEntries(entries []*common.RegistrationEntry) []*comm
 	return []*common.RegistrationEntry(e)
 }
 
-func newPlugin(dbType string) (datastore.DataStore, error) {
-	db, err := gorm.Open("sqlite3", dbType)
-	if err != nil {
-		return nil, err
+// restart will close and re-open the sqlite database.
+func (ds *sqlitePlugin) restart() error {
+	ds.mutex.Lock()
+	defer ds.mutex.Unlock()
+
+	// Build sqlite connect string
+	path := ds.fileName
+	if path == ":memory:" {
+		path = path + "?cache=shared"
 	}
+	path = "file:" + path
 
-	db.LogMode(true)
+	log.Printf("opening sqlite database with path %s", path)
+	db, err := gorm.Open("sqlite3", path)
```

Comment:
In case of a file path, I think that we will now need to check if the folder exists.
I guess that there will be an error here if you don't have the .data folder. We can automatically create the folder in that case at this point.

Reply:
I'm hesitant to create directories if they don't exist, since it might be an error condition rather than an omission.

How about I create an empty `.data` dir and check it in?

Reply:
💯 Yes, that's better.

Reply:
@amartinezfayo fixed. Git automatically removes empty directories, so I put a placeholder file there.